src/libraries/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Generators.Tests/RegexGeneratorParserTests.cs

```diff
@@ -556,6 +556,46 @@ partial class C
                 }", compile: true, additionalRefs: new[] { MetadataReference.CreateFromImage(referencedAssembly) }));
         }
 
+        [Fact]
+        public async Task WithGroups_ShouldDisplayGroupName()
+        {
+            string code = @"
+                using System.Text.RegularExpressions;
+                partial class C1
+                {
+                    [RegexGenerator(""^(?<LINE>[0-9]*)-(?<ENDLINE>[0-9]*)$"")]
+                    private static partial Regex A();
+                }
+            ";
+            LanguageVersion langVersion = LanguageVersion.Preview;
+            var proj = new AdhocWorkspace()
+               .AddSolution(SolutionInfo.Create(SolutionId.CreateNewId(), VersionStamp.Create()))
+               .AddProject("RegexGeneratorTest", "RegexGeneratorTest.dll", "C#")
+               .WithMetadataReferences(s_refs)
+               .WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
+               .WithNullableContextOptions(NullableContextOptions.Enable))
+               .WithParseOptions(new CSharpParseOptions(langVersion))
+               .AddDocument("RegexGenerator.g.cs", SourceText.From(code, Encoding.UTF8)).Project;
+
+            Assert.True(proj.Solution.Workspace.TryApplyChanges(proj.Solution));
+
+            Compilation? comp = await proj!.GetCompilationAsync(CancellationToken.None).ConfigureAwait(false);
+            Debug.Assert(comp is not null);
+
+            var generator = new RegexGenerator();
+            CSharpGeneratorDriver cgd = CSharpGeneratorDriver.Create(new[] { generator.AsSourceGenerator() }, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(langVersion));
+            GeneratorDriver gd = cgd.RunGenerators(comp!);
+            GeneratorDriverRunResult generatorResults = gd.GetRunResult();
+            Assert.NotNull(generatorResults);
+            GeneratorRunResult generatorRunResult = Assert.Single(generatorResults.Results);
+            GeneratedSourceResult generatedSourceResult = Assert.Single(generatorRunResult.GeneratedSources);
+            string sourceText = generatedSourceResult.SourceText.ToString();
+            Assert.Contains("// 1st capturing group (\"LINE\").", sourceText);
+            Assert.Contains("// 2nd capturing group (\"ENDLINE\").", sourceText);
+            Assert.Contains("// \u25CB 1st capturing group (\"LINE\").", sourceText);
+            Assert.Contains("// \u25CB 2nd capturing group (\"ENDLINE\").", sourceText);
+        }
```

Comment:
I applaud your adding a test.  However, I don't want to duplicate all of this workspace/generator/compilation logic: if the existing helper is insufficient to perform the desired validation, we should refactor it to be suitable.  We also don't currently have tests that validate any other aspects of code generation.  I believe we have an open issue to do that, but regardless, @joperezr has it on his plate to add a bunch of tests that validate the code generated for a given set of input expressions is exactly what we expect, and that will naturally handle this case as well as long as the input expressions use named capture groups (part of his work will be ensuring we have good coverage of all features in the tested expressions). Thus, while normally I encourage adding unit tests, I'm ok with not having one in this PR.